### ISSUE TITLE
Clearing product page cache after product purchased and stock decreased

### ISSUE THREAD

**Issue by john-henry** [putyourlightson/craft-blitz#432]
Using Craft Commence and on a product page I use similar to 

`{% do craft.blitz.options.tags(['groupId:5','typeId:14']) %}`

We use stock on the page with various bits of logic.

if the product is updated etc then Blitz cache gets cleared as per the tag above so all good.

However if somebody purchases the item then the stock level goes down but Blitz does not clear the cache. Even though essentially the product has been updated. This is causing us some trouble. Is this a bug, a FR or just not supported?

**Comment by bencroker** [comment]
This came up a while back in https://github.com/putyourlightson/craft-blitz/issues/120, but it might be worth considering adding this to Blitz directly, as an integration. Here is how you might solve it with a custom module/plugin in the meantime.

```php

Event::on(Order::class, Order::EVENT_AFTER_COMPLETE_ORDER,
    function(Event $event) {
        /** @var Order $order */
        $order = $event->sender;
        foreach ($order->getLineItems() as $lineItem) {
            $purchasable = $lineItem->getPurchasable();
            if ($purchasable instanceof Element) {
                Blitz::$plugin->refreshCache->addElement($purchasable);
            }
        }
        Blitz::$plugin->refreshCache->refresh();
    }
);
```

**Comment by john-henry** [comment]
Ah OK. I swear I searched issues first! Ill give this a go and report back

**Comment by john-henry** [comment]
This works. Thanking you

**Comment by bencroker** [comment]
Great, keeping this open as a FR.

**Comment by bencroker** [comment]
Added in https://github.com/putyourlightson/craft-blitz/commit/8aefccf90801d0fcc5534072b623823c7958c6ca for the next release. If you could help test, that would be much appreciated.

**Comment by bencroker** [comment]
Released in version 4.2.0.
